File: jcv-core/src/main/java/com/ekino/oss/jcv/core/comparator/NotEmptyComparator.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator;

import java.util.Objects;

import com.ekino.oss.jcv.core.JsonValueComparator;

import org.skyscreamer.jsonassert.ValueMatcherException;

public class NotEmptyComparator implements JsonValueComparator<String> {

    @Override
    public boolean hasCorrectValue(String actual, String expected) {
        if (actual != null && !actual.isEmpty()) {
            return true;
        }
        throw new ValueMatcherException("Value should not be empty", Objects.toString(expected), Objects.toString(actual));
    }
}

File: jcv-core/src/main/kotlin/com/ekino/oss/jcv/core/comparator/NotEmptyComparator.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator

import com.ekino.oss.jcv.core.JsonValueComparator
import org.skyscreamer.jsonassert.ValueMatcherException
import java.util.Objects

class NotEmptyComparator : JsonValueComparator<String> {

  override fun hasCorrectValue(actual: String?, expected: String?): Boolean {
    if (actual.isNullOrEmpty().not()) {
      return true
    }
    throw ValueMatcherException("Value should not be empty", Objects.toString(expected), Objects.toString(actual))
  }
}
